$OpenBSD: patch-CMakeLists_txt,v 1.1 2019/06/22 20:18:00 rsadowski Exp $

Fix build with Ninja 1.9.0.
Create only one custom target to generate config.h to avoid duplicate ninja
rules.

Index: CMakeLists.txt
--- CMakeLists.txt.orig
+++ CMakeLists.txt
@@ -40,6 +40,16 @@ INCLUDE( ${QT_USE_FILE} )
 
 ADD_DEFINITIONS(${QT_DEFINITIONS})
 
+# configure the header config.h
+CONFIGURE_FILE (
+    "${PROJECT_SOURCE_DIR}/config.h.cmake"
+    "${PROJECT_SOURCE_DIR}/config.h"
+)
+SET_SOURCE_FILES_PROPERTIES(${PROJECT_SOURCE_DIR}/config.h PROPERTIES GENERATED 1)
+ADD_CUSTOM_TARGET(
+    generate_config_header
+    DEPENDS ${PROJECT_SOURCE_DIR}/config.h
+)
 
 ADD_SUBDIRECTORY( qucs )
 ADD_SUBDIRECTORY( qucs-activefilter )
